components/mediation-connector/org.wso2.carbon.mediation.connector.as4/src/main/java/org/wso2/carbon/mediation/connector/message/util/AS4Utils.java

```diff
@@ -0,0 +1,136 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.mediation.connector.message.util;
+
+import org.apache.axiom.om.OMNode;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.mediation.connector.exceptions.AS4ErrorMapper;
+import org.wso2.carbon.mediation.connector.exceptions.AS4Exception;
+import org.wso2.carbon.mediation.connector.message.beans.Messaging;
+import org.wso2.carbon.mediation.connector.pmode.PModeRepository;
+import org.wso2.carbon.mediation.connector.pmode.impl.PMode;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.PropertyException;
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+public class AS4Utils {
+
+    private static final Log log = LogFactory.getLog(AS4Utils.class);
+
+    public static OMNode getOMNode(final Marshaller marshaller, final Messaging messaging)
```

Comment:
Added in https://github.com/wso2/carbon-mediation/pull/842/commits/943cc42afba8e08bec9dc5ef78b4df9664b688df